src/lib/transition_frontier/frontier_base/root_data.ml

```diff
@@ -121,6 +121,13 @@ module Limited = struct
   let scan_state t = Common.scan_state t.common
 
   let pending_coinbase t = Common.pending_coinbase t.common
+
+  let read_all_proofs_from_disk t =
```

Comment:
Why this type could only be read from disk? Moreover, could you explain a bit how are these 3 different root data types being used in our codebase?